### PR TITLE
Affiliation Program Auto Enrollment in EDA Settings Not Respecting API Name for Account Record Type

### PR DESCRIPTION
**Pull request by MatthewBlanski**
# Critical Changes

# Changes

* We fixed a problem that made Auto Enrollment not work when an Account Record Type API name was used in an Auto Enrollment Mapping.

# Issues Closed

# New Metadata

## Unpackaged Metadata

# Deleted Metadata

# Testing Notes

### CODE REVIEW

File: force-app/main/default/classes/AFFL_MultiRecordType_TEST.cls [hunk 1]
```diff
@@ -36,336 +36,11 @@
  */
 @isTest
 public with sharing class AFFL_MultiRecordType_TEST {
-    /**************************************************************************
```
Comment (MatthewBlanski) [inline force-app/main/default/classes/AFFL_MultiRecordType_TEST.cls:39]: Extracted to force-app/main/default/classes/AffiliationTestSetupFactory_TEST

File: force-app/main/default/classes/AFFL_MultiRecordType_TEST.cls [hunk 2]
```diff
@@ -1374,377 +961,19 @@ public with sharing class AFFL_MultiRecordType_TEST {
...
-
-        Test.startTest();
-        AFFL_MultiRecordTypeMapper afflMapper = new AFFL_MultiRecordTypeMapper();
-        String primaryAfflLookupAPIName = afflMapper.getPrimaryAffiliationLookupAPI('Test1234');
-        Test.stopTest();
-
-        System.assertEquals(null, primaryAfflLookupAPIName);
-    }
-
-    /**************************************************************************
```
Comment (MatthewBlanski) [inline force-app/main/default/classes/AFFL_MultiRecordType_TEST.cls:1568]: Extracted from here to force-app/main/default/classes/AffiliationTestSetupFactory_TEST

File: force-app/main/default/classes/AFFL_MultiRecordType_TEST.cls [hunk 3]
```diff
@@ -1100,101 +777,12 @@ public with sharing class AFFL_MultiRecordType_TEST {
         System.assertEquals(null, Contact.Primary_Organization__c);
     }
 
-    // Test getPrimaryAffiliationLookupAPIFromAffiliationType method of AFFL_MultiRecordTypeMapper
```
Comment (MatthewBlanski) [inline force-app/main/default/classes/AFFL_MultiRecordType_TEST.cls:1103]: Extracted to force-app/main/default/classes/AFFL_MultiRecordTypeMapper_TEST.cls

File: force-app/main/default/classes/AFFL_MultiRecordType_TEST.cls [hunk 4]
```diff
@@ -1374,377 +961,19 @@ public with sharing class AFFL_MultiRecordType_TEST {
...
 
+        Test.startTest();
+        insert affiliationsToBeInserted;
         Test.stopTest();
 
         // Ensure 4 new Affiliations are inserted
         List<Affiliation__c> affiliationsInserted = [SELECT Id FROM Affiliation__c];
         System.assertEquals(5, affiliationsInserted.size());
     }
-
```
Comment (MatthewBlanski) [inline force-app/main/default/classes/AFFL_MultiRecordType_TEST.cls:1391]: Extracted to force-app/main/default/classes/AFFL_MultiRecordTypeMapper_TEST.cls

**Review by xinnanlisf — APPROVED** [review]

**Pull request merged**